src/Core/Compiler/NodeConverter.php

```diff
@@ -107,12 +107,14 @@ public function convert(NodeInterface $node)
     protected function convertEscapingNode(EscapingNode $node)
     {
         $configuration = $this->convert($node->getNode());
-        $configuration['execution'] = sprintf(
-            'call_user_func_array( function ($var) { ' .
-            'return (is_string($var) || (is_object($var) && method_exists($var, \'__toString\')) ' .
-            '? htmlspecialchars((string) $var, ENT_QUOTES) : $var); }, [%s])',
-            $configuration['execution']
-        );
+        if (!empty($configuration['execution']) && strtolower($configuration['execution']) !== 'null' && !is_numeric(trim($configuration['execution'], '\'"'))) {
```

Comment:
this has been changed for empty string returns with 9785955cf59014dcba234f921d5d150bfdc1eec9
already. further changes depend on booleanNode convert() to be changed.